virttest/utils_time.py

```diff
@@ -0,0 +1,248 @@
+    timezone_pattern = '^(?:\s+Time zone:\s)(\w+\/\S+|UTC)(?:\s\(\S+,\s)([+|-]\d{4})\)$'
+    error_context.context("Get host's timezone", logging.info)
+    host_timezone = process.system_output(timezone_cmd, timeout=240, shell=True)
+    try:
+        host_timezone_set = re.match(timezone_pattern, host_timezone).groups()
+        return {"timezone_city": host_timezone_set[0],
+                "timezone_code": host_timezone_set[1]}
+    except AttributeError, IndexError:
+        raise exceptions.TestError("Fail to get host's timezone.")
+
+
+def verify_timezone(session):
+    """
+    Verify linux guest's timezone.
+
+    :param session: VM session.
+    """
+    error_context.context("Verify guest's timezone", logging.info)
+    timezone_cmd = 'timedatectl | grep "Time zone"'
+    timezone_pattern = '^(?:\s+Time zone:\s)(\w+\/\S+|UTC)(?:\s\(\S+,\s)([+|-]\d{4})\)$'
+    guest_timezone = session.cmd_output(timezone_cmd, timeout=240)
+    try:
+        guest_timezone_set = re.match(timezone_pattern, guest_timezone).groups()
+        return (guest_timezone_set[0] == get_host_timezone()['timezone_city'])
+    except AttributeError, IndexError:
+        raise exceptions.TestError("Fail to get guest's timezone.")
+
+
+def sync_timezone(session):
+    """
+    Sync linux guest's timezone
+
+    :param session: VM session
+    """
+    error_context.context("Sync guest's timezone", logging.info)
+    set_timezone_cmd = "timedatectl set-timezone %s"
+    if not verify_timezone(session):
+        host_timezone_city = get_host_timezone()['timezone_city']
+        session.cmd(set_timezone_cmd % host_timezone_city)
+        if not verify_timezone(session):
+            raise exceptions.TestError("Fail to sync guest's timezone.")
+    return True
+
+
+def verify_timezone_win(session):
+    """
+    Verify windows guest's timezone
+
+    :params session: VM session
+    """
+    def get_timezone_list():
+        timezone_list_cmd = "tzutil /l"
+        timezone_set = []
+        timezone_sets = []
+        timezone_list = session.cmd_output(timezone_list_cmd)
+
+        for line in timezone_list.splitlines():
+            # empty line
+            if not line:
+                continue
+            _code = re.search(r'(?:\(UTC([+|-]\d{2}:\d{2})?)', line)
+            if _code and _code.group(1):
+                _code_ = re.sub(r'(\d{2}):(\d{2})', r'\1\2',
+                                _code.group(1))
+                timezone_set.append(_code_)
+                continue
+            # UTC standard time, add timezone code '+0000'.
+            elif _code:
+                timezone_set.append("+0000")
+                continue
+            _name = re.search(r'(\S+(?:\s\S+)*)', line)
+            if _name:
+                timezone_set.append(_name.group(0))
+            else:
+                logging.warn("Can not get timezone name correctly.")
+            timezone_sets.append(timezone_set)
+            timezone_set = []
```

Comment:
what does `timezone_set` contain? It seems its length will always be 1. 
When do you want to `timezone_set` to be an empty list?
If you want to make sure `timezone_set = []` in the `for` loop, it seems you can remove line 68 and 92, put `timezone_set = []` after line 72.

Reply:
Thanks. `timezone_set` is a set(not the python set) of timezone, such as `["+1200", "Fiji Standard Time"]`. It can get second value when program goes to `continue` in line80 or line84, and a new line will be in `line`. 
The format in `timezone_list` is:

>(UTC+12:00) Fiji
Fiji Standard Time
(empty line)
(UTC) Coordinated Universal Time
UTC

And `timezone_sets` is a list(python list) contains all the sets of timezone, like `[["+1200", "Fiji Standard Time"], ["-0100", "Azores Standard Time"],...]`.